### ISSUE TITLE
Completion does not replace existing text.

### ISSUE THREAD

**Issue by acristoffers** [zigtools/zls#907]
For clients that support the ApplyEdit capability, the code completion can return a range that allows the editor to replace text instead of only inserting new text. For example, in the case where the completion is requested as in:

```zig
const varA = 0;
const varB = 1;
const a = var<cursor>A;
```

the completion as of now will just insert the new text, leaving everything that comes after the cursor:

```zig
const a = varAA;
const a = varBA;
```

but it should replace the whole identifier with the new value:

```zig
const a = varA;
const a = varB;
```

**Comment by Techatrix** [comment]
The partial fix for this issue mentioned the following:
> TODO Identify function call/struct init and replace the whole thing.

is this why this issue is still open?

**Comment by Techatrix** [comment]
No response, so I will consider this fixed by #1795
